File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/3738-MakeArrayNonDecreasing/3738-MakeArrayNonDecreasing.cpp
// Last updated: 8/4/2026, 12:35:15 PM
class Solution {
public:
    int maximumPossibleSize(vector<int>& nums) {
        int size = 0, prev = -1;

        for (int num : nums) {
            if (num >= prev) {
                prev = num;
                size++;
            }
        }

        return size;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/3738-MakeArrayNonDecreasing/3738-MakeArrayNonDecreasing.java
// Last updated: 8/4/2026, 12:35:22 PM
class Solution {
    public int maximumPossibleSize(int[] nums) {
        int n = nums.length;
        Stack<Integer>st = new Stack<>();

        for(int i = 0 ; i < n ; i++){
            while(!st.empty() && nums[i] < st.peek()){
                nums[i] = Math.max(st.pop() , nums[i]);
            }
            st.push(nums[i]);
        }
        return st.size();
    }
}